### PR TITLE
Updates and bug fixes to how-to-segment-anything-with-fast-sam.ipynb

### PR DESCRIPTION
**Pull request by MattLondon101**
# Description

### CHANGE 1: Changed typo in Markdown cell: 

**NOTE:** `FastSAM` code is not distributed via `pip` not it is packaged. Make sure to run code b**a**low from `{HOME}/FastSAM` directory. ⚠️

to 

**NOTE:** `FastSAM` code is not distributed via `pip` not it is packaged. Make sure to run code b**e**low from `{HOME}/FastSAM` directory. ⚠️
#
### CHANGE 2: Added print statement for DEVICE:
`print(f"DEVICE = {DEVICE}")`
#
### CHANGE 3: Added cell to create output directory and path:
`os.makedirs(f"{HOME}/output", exist_ok=True)`
`output_image_path = f"{HOME}/output/output-{os.path.basename(IMAGE_PATH)}"`   
 
This prevents error: `OpenCV(4.9.0) /io/opencv/modules/imgcodecs/src/loadsave.cpp:696: error: (-2:Unspecified error) could not find a writer for the specified extension in function 'imwrite_'` when running `prompt_process.plot(annotations=masks, output_path=f"{HOME}/output")`
#
### CHANGE 4: Changed **NOTE:** regarding output for prompt_process.everything_prompt:

**NOTE:** `prompt_process.everything_prompt` returns `torch.Tensor`

to

**NOTE:** `prompt_process.everything_prompt` returns `torch.Tensor` when DEVICE = 'cuda:0'. `prompt_process.everything_prompt` returns `numpy.ndarray` when DEVICE = 'cpu'.
#
### CHANGE 5: Added function `masks_to_bool(masks)`:  
Converts masks to boolean, regardless of if masks are numpy.ndarray or torch.Tensor
#
### CHANGE 6: Replaced `mask_annotator = sv.MaskAnnotator()` with `mask_annotator = sv.MaskAnnotator(color_lookup = sv.ColorLookup.INDEX)`  

- Prevents error: `ValueError: Could not resolve color by class becauseDetections do not have class_id` when running `annotated_image=annotate_image(image_path=IMAGE_PATH, masks=masks)`
- Fix found at [https://github.com/roboflow/supervision/discussions/428](https://github.com/roboflow/supervision/discussions/428)
#
Please include a summary of the change and which issue is fixed or implemented. Please also include relevant motivation and context (e.g. links, docs, tickets etc.).

- Changes are a result of my personal demo of `how-to-segment-anything-with-fast-sam.ipynb` and running into the issues described in each CHANGE.
- Updates to account for running in CPU are due to wanting to run notebook in cases where GPU not available.
#
List any dependencies that are required for this change.

- Python version: 3.10.14
- No additional dependencies beyond what was already listed in imports
#
## Type of change

Please delete options that are not relevant.

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] This change requires a documentation update
#
## How has this change been tested, please provide a testcase or example of how you tested the change?

- Ran entire `update--how-to-segment-anything-with-fast-sam.ipynb` successfully in Google Colab with CPU and T4 GPU
- Ran entire `update--how-to-segment-anything-with-fast-sam.ipynb` successfully in Ubuntu 22.04 in Conda environment, Python 3.10.14, with CPU 

## Any specific deployment considerations

For example, documentation changes, usability, usage/costs, secrets, etc.:
- Orgingal notebook, `how-to-segment-anything-with-fast-sam.ipynb` may need to be replace with this updated notebook (`update--how-to-segment-anything-with-fast-sam.ipynb`).

## Docs

-   [ ] Docs updated? What were the changes:
- No Docs changes beyond some Markdown changes in notebook (described in CHANGES above).

### CODE REVIEW

**Comment by SkalskiP** [comment]
Hi, @MattLondon101 👋🏻 Thanks for your interest in notebooks and the time you spent working on this PR. Could you update the PR and commit your notebook under the original name? We wouldn't like to have two versions of the same notebook, we would like to update the original one.

**Comment by MattLondon101** [comment]
@SkalskiP I have updated the PR with latest commit: "Deleted notebooks/how-to-seg
ment-anything-with-fast-sam.ipynb. Then renamed notebooks/update--how-to-segment-anything-with-fast-sam.ipynb to notebooks/how-to-segment-anything-with-fast-sam.ipynb"

**Comment by SkalskiP** [comment]
Hi @MattLondon101 👋🏻 Could you accept the CLA?

**Comment by MattLondon101** [comment]
@SkalskiP I have accepted the CLA. Please let me know if there is anything else I can do.

File: README.md [hunk 1]
```diff
@@ -73,7 +73,7 @@ Curious to learn more about GPT-4 Vision? [Check out our GPT-4V experiments 🧪
...
...ooks/blob/main/notebooks/zero-shot-object-detection-with-yolo-world.ipynb) | [![Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/roboflow-ai/notebooks/blob/main/notebooks/zero-shot-object-detection-with-yolo-world.ipynb) [![Kaggle](https://kaggle.com/static/images/open-in-kaggle.svg)](https://kaggle.com/kernels/welcome?src=https://github.com/roboflow-ai/notebooks/blob/main/notebooks/zero-shot-object-detection-with-yolo-world.ipynb) [![SageMaker](https://raw.githubusercontent.com/roboflow-ai/notebooks/main/assets/badges/sage-maker.svg)](https://studiolab.sagemaker.aws/import/github/roboflow-ai/notebooks/blob/main/notebooks/zero-shot-object-detection-with-yolo-world.ipynb) | [![Roboflow](https://raw.githubusercontent.com/roboflow-ai/notebooks/main/assets/badges/roboflow-blogpost.svg)](https://blog.roboflow.com/what-is-yolo-world/) [![YouTube](https://badges.aleen42.com/src/youtube.svg)](https://www.youtube.com/watch?v=X7gKBGVz4vs) | [![GitHub](https://badges.aleen42.com/src/github.svg)](https://github.com/AILab-CVC/YOLO-World) [![arXiv](https://img.shields.io/badge/arXiv-2401.1727-b31b1b.svg)](https://arxiv.org/abs/2401.1727)|
```
Comment (SkalskiP) [inline README.md:76]: It looks like you updated the ID of the arXiv paper. https://arxiv.org/abs/2401.1727 is to YOLO-World paper.

File: README.md [hunk 2]
```diff
@@ -99,10 +99,10 @@ Curious to learn more about GPT-4 Vision? [Check out our GPT-4V experiments 🧪
...
+| [YOLOR Object Detection](https://github.com/roboflow-ai/notebooks/blob/main/notebooks/train-yolor-object-detection-on-custom-data.ipynb) | [![Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/roboflow-ai/notebooks/blob/main/notebooks/train-yolor-object-detection-on-custom-data.ipynb) [![Kaggle](https://kaggle.com/static/images/open-in-kaggle.svg)](https://kaggle.com/kernels/welcome?src=https://github.com/roboflow-ai/notebooks/blob/main/notebooks/train-yolor-object-detection-on-custom-data.ipynb)  | [![Roboflow](https://raw.githubusercontent.com/roboflow-ai/notebooks/main/assets/badges/roboflow-blogpost.svg)](https://blog.roboflow.com/train-yolor-on-a-custom-dataset) [![YouTube](https://badges.aleen42.com/src/youtube.svg)](https://www.youtube.com/watch?v=sZ5DiXDOHEM) | [![GitHub](https://badges.aleen42.com/src/github.svg)](https://github.com/roboflow-ai/yolor) [![arXiv](https://img.shields.io/badge/arXiv-1506.0264-b31b1b.svg)](https://arxiv.org/abs/1506.0264)|
```
Comment (SkalskiP) [inline README.md:102]: Same here. https://arxiv.org/abs/1506.0264 and https://arxiv.org/abs/2107.0843 are wrong papers.

**Comment by SkalskiP** [comment]
Hi @MattLondon101 👋🏻 Let's clean up a few minor things before we merge.

- Please remove all changes you made in `README.md` and `automation/notebooks-table-data.csv` files.
- Looks like this line `output_image_path = f"{HOME}/output/output-{os.path.basename(IMAGE_PATH)}"` is redundant. `output_image_path` is not used anywhere. Please remove it.
It looks like we could simplify the `masks_to_bool` implementation

```python
def masks_to_bool(masks):
    if type(masks) == np.ndarray:
        return masks.astype(bool)
    return masks.cpu().numpy().astype(bool)
```

**Review by SkalskiP — CHANGES_REQUESTED** [review]

**Comment by MattLondon101** [comment]
@SkalskiP I've made all changes requested. I think the README.md and notebooks-table-data.csv got messed up when I was initially thinking to make a new notebook for the update, instead of replacing the original one. I like your simplification of `masks_to_bool`. Please let me know if any more changes needed.

**Comment by SkalskiP** [comment]
@MattLondon101, no worries ;) I'm just making sure everything works. Merging! Thanks a lot for the contribution! 🙏🏻

**Review by SkalskiP — APPROVED** [review]

**Pull request merged**